src/com/goide/inspections/GoReservedWordUsedAsNameInspection.java

```diff
@@ -34,57 +35,56 @@
   protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNull LocalInspectionToolSession session) {
     return new GoVisitor() {
       @Override
-      public void visitFunctionDeclaration(@NotNull GoFunctionDeclaration o) {
+      public void visitTypeSpec(@NotNull GoTypeSpec o) {
+        super.visitTypeSpec(o);
         check(o, holder);
       }
 
       @Override
-      public void visitVarDefinition(@NotNull GoVarDefinition o) {
+      public void visitConstDefinition(@NotNull GoConstDefinition o) {
+        super.visitConstDefinition(o);
         check(o, holder);
       }
-    };
-  }
-
-  private static void check(@NotNull GoFunctionDeclaration function, @NotNull ProblemsHolder holder) {
-    GoFile builtin = GoSdkUtil.findBuiltinFile(function);
-    if (builtin == null) return;
 
-    String name = function.getName();
-    if (name == null) return;
+      @Override
+      public void visitFunctionOrMethodDeclaration(@NotNull GoFunctionOrMethodDeclaration o) {
+        super.visitFunctionOrMethodDeclaration(o);
+        check(o, holder);
+      }
 
-    for (GoFunctionDeclaration builtinFunctionDeclaration : builtin.getFunctions()) {
-      if (name.equals(builtinFunctionDeclaration.getName())) {
-        registerProblem(holder, function, builtinFunctionDeclaration, name);
-        break;
+      @Override
+      public void visitVarDefinition(@NotNull GoVarDefinition o) {
+        super.visitVarDefinition(o);
+        check(o, holder);
       }
-    }
+    };
   }
 
-  private static void check(@NotNull GoVarDefinition variable, @NotNull ProblemsHolder holder) {
-    GoFile builtin = GoSdkUtil.findBuiltinFile(variable);
+  private static void check(@NotNull GoNamedElement element, @NotNull ProblemsHolder holder) {
+    ProgressManager.checkCanceled();
+    GoFile builtin = GoSdkUtil.findBuiltinFile(element);
     if (builtin == null) return;
 
-    String name = variable.getName();
-    if (name == null) return;
+    String name = element.getName();
+    if (name == null || "Type".equals(name)) return;
```

Comment:
`Type` is there because it's a rather special case in the `builtin.go` file. However, I'm starting to to think a different approach should be considered since there are other cases as well.

See: https://github.com/golang/go/blob/master/src/builtin/builtin.go#L105